File: sdk/julia/src/url_shortener.jl
# URL Shortener: shorten URLs, manage links, stats
struct UrlShortenerResource
    http::HttpClient
end

# POST /api/url-shortener/shorten - Shorten a URL
function shorten(r::UrlShortenerResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/url-shortener/shorten"; params = params)
end

# GET /api/url-shortener/links - List links
function list_links(r::UrlShortenerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/url-shortener/links"; params = params)
end

# GET /api/url-shortener/config - Get shortener config
function get_config(r::UrlShortenerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/url-shortener/config"; params = params)
end

# GET /api/url-shortener/stats - Get link stats
function get_stats(r::UrlShortenerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/url-shortener/stats"; params = params)
end

# GET /api/url-shortener/links/{id} - Get link details
function get_link(r::UrlShortenerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/url-shortener/links/{id}"; params = params)
end

# DELETE /api/url-shortener/links/{id} - Delete link
function delete_link(r::UrlShortenerResource; kwargs...)
    params = Dict(kwargs)
    http_delete(r.http, "/api/url-shortener/links/{id}"; params = params)
end

File: sdk/rust/src/url_shortener.rs
//! URL Shortener: shorten URLs, manage links, stats

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct UrlShortenerResource {
    http: Arc<HttpClient>,
}

impl UrlShortenerResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// POST /api/url-shortener/shorten - Shorten a URL
    pub async fn shorten(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/url-shortener/shorten", None).await
    }

    /// GET /api/url-shortener/links - List links
    pub async fn list_links(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/url-shortener/links", None).await
    }

    /// GET /api/url-shortener/config - Get shortener config
    pub async fn get_config(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/url-shortener/config", None).await
    }

    /// GET /api/url-shortener/stats - Get link stats
    pub async fn get_stats(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/url-shortener/stats", None).await
    }

    /// GET /api/url-shortener/links/{id} - Get link details
    pub async fn get_link(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/url-shortener/links/{id}", None).await
    }

    /// DELETE /api/url-shortener/links/{id} - Delete link
    pub async fn delete_link(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.delete("/api/url-shortener/links/{id}", None).await
    }

}
